jurity/recommenders/interlist_diversity.py

```diff
@@ -0,0 +1,214 @@
+# -*- coding: utf-8 -*-
+# Copyright FMR LLC <opensource@fidelity.com>
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import List, Union, Tuple
+
+import numpy as np
+import pandas as pd
+import scipy.sparse as sp
+from sklearn.metrics.pairwise import cosine_similarity
+import warnings
+
+from jurity.utils import Constants
+
+
+def sample_users(df: pd.DataFrame, user_id_column: str = Constants.user_id,
+                 sample_size: float = None, seed: int = Constants.default_seed) -> pd.DataFrame:
+    """
+    Samples input data frame by selecting a random sample of users.
+
+    Parameters
+    ----------
+    df: pd.DataFrame
+        Data frame with a user_id_col column.
+    user_id_column: str
+        User id column name.
+    sample_size: float
+        Proportion of users to randomly sample for evaluation.
+    seed : int, default=Constants.default_seed
+        The seed used to create random state.
+
+    Returns
+    -------
+    Sampled data frame
+    """
+    rng = np.random.RandomState(seed)
+    users = df[user_id_column].unique()
+    users = rng.choice(users, size=int(len(users) * sample_size), replace=False)
+    return df[df[user_id_column].isin(users)]
+
+
+def tocsr(df: pd.DataFrame, user_id_column: str = Constants.user_id, item_id_column: str = Constants.item_id):
```

Comment:
Nice, I think I've written similar pieces of code before, and I thought it was a shame it was not readily available. So following that, I think it would be nice to move this to `utils` so others can import and use it.

Reply:
if this moves to utils, I don't see what id_column and item_column str means for a to_csr() function

Reply:
This is a `tocsr` method specifically for reco cases, when we have user/item/rating tuples. So the user and item column strings should be relevant when using it.

Reply:
If move this to utils, I wouldn't object to update the params to be `row_id` and `col_id` I guess. In this context, we are using users as rows and items as columns.

Reply:
sounds like a good generalization, and then the reco part specializes in what rows, column stand for. Cool!